### PR TITLE
Export .scss alongside .css

### PR DESCRIPTION
**Pull request by eFrane**
This is a feeble and long-since in the pipes attempt to export both the scss file and the css file from the css extraction as discussed in #420. Original discussion on this design path [starts here](https://github.com/shentao/vue-multiselect/issues/420#issuecomment-427690620).

As stated in d18b25c, I could not quite make it to the finish line. Hopefully, there's a webpack magician around to get this done. 

Note: Kudos to @assembledadam and @jeremykenedy for having done the prior conversion work to scss.

### CODE REVIEW

File: src/Multiselect.vue [hunk 1]
```diff
@@ -143,689 +143,738 @@
...
-  from {
-    transform: rotate(0);
-  }
-  to {
-    transform: rotate(2turn);
-  }
-}
+<style lang="scss">
+    $multiselect-color-brand-light: #41b883;
+    $multiselect-color-brand-dark: darken(#41b883, 20%);
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
    $multiselect-color-brand-dark: darken($multiselect-color-brand-light, 20%);
```

File: src/Multiselect.vue [hunk 2]
```diff
@@ -143,689 +143,738 @@
...
+    }
+
+    .multiselect__input:focus {
+        border-color: $multiselect-color-border-focus;
+        outline: none;
+    }
+
+    .multiselect__single {
+
+        &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 3]
```diff
@@ -143,689 +143,738 @@
...
+        :focus {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
        &:focus {
```

File: src/Multiselect.vue [hunk 4]
```diff
@@ -143,689 +143,738 @@
...
+        right: 1px;
+        top: 1px;
+        padding: 4px 8px;
+        margin: 0;
+        text-decoration: none;
+        text-align: center;
+        cursor: pointer;
+        transition: transform 0.2s ease;
+
+        &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 5]
```diff
@@ -143,689 +143,738 @@
...
+        :before {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
        &:before {
```

File: src/Multiselect.vue [hunk 6]
```diff
@@ -143,689 +143,738 @@
...
+        min-height: 40px;
+        line-height: 16px;
+        text-decoration: none;
+        text-transform: none;
+        vertical-align: middle;
+        position: relative;
+        cursor: pointer;
+        white-space: nowrap;
+
+        &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 7]
```diff
@@ -143,689 +143,738 @@
...
+        :after {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
        &:after {
```

File: src/Multiselect.vue [hunk 8]
```diff
@@ -143,689 +143,738 @@
...
+        }
+
+    }
+
+    .multiselect__option--highlight {
+        background: $multiselect-color-background-active;
+        outline: none;
+        color: $multiselect-color-text-active;
+
+        &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 9]
```diff
@@ -143,689 +143,738 @@
...
+        :after {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
        &:after {
```

File: src/Multiselect.vue [hunk 10]
```diff
@@ -143,689 +143,738 @@
...
+        }
+
+    }
+
+    .multiselect__option--selected {
+        background: $multiselect-color-background-group-selected;
+        color: $multiselect-color-text;
+        font-weight: bold;
+
+        &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 11]
```diff
@@ -143,689 +143,738 @@
...
+        :after {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
        &:after {
```

File: src/Multiselect.vue [hunk 12]
```diff
@@ -143,689 +143,738 @@
...
+            content: attr(data-selected);
+            color: silver;
+        }
+
+        &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 13]
```diff
@@ -143,689 +143,738 @@
...
+        .multiselect__option--highlight {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
        &.multiselect__option--highlight {
```

File: src/Multiselect.vue [hunk 14]
```diff
@@ -143,689 +143,738 @@
...
+            background: $multiselect-color-group;
+            color: $multiselect-color-text-active;
+
+            &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 15]
```diff
@@ -143,689 +143,738 @@
...
+            :after {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
            &:after {
```

File: src/Multiselect.vue [hunk 16]
```diff
@@ -143,689 +143,738 @@
...
+    .multiselect__option--group {
+        background: $multiselect-color-background-disabled;
+        color: $multiselect-color-text;
+
+        &
+        .multiselect__option--highlight {
+            background: $multiselect-color-background-group-highlight;
+            color: $multiselect-color-text-active;
+
+            &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 17]
```diff
@@ -143,689 +143,738 @@
...
+
+    .multiselect__option--disabled.multiselect__option--highlight {
+        background: $multiselect-color-background-group-disabled;
+    }
+
+    .multiselect__option--group-selected.multiselect__option--highlight {
+        background: $multiselect-color-group;
+        color: $multiselect-color-text-active;
+
+        &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 18]
```diff
@@ -143,689 +143,738 @@
...
+        color: $multiselect-color-group-text-disabled !important;
+        cursor: text;
+        pointer-events: none;
+    }
+
+    .multiselect__option--group {
+        background: $multiselect-color-background-disabled;
+        color: $multiselect-color-text;
+
+        &
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
```

File: src/Multiselect.vue [hunk 19]
```diff
@@ -143,689 +143,738 @@
...
+        .multiselect__option--highlight {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
        &.multiselect__option--highlight {
```

File: src/Multiselect.vue [hunk 20]
```diff
@@ -143,689 +143,738 @@
...
+            background: $multiselect-color-background-group-highlight;
+            color: $multiselect-color-text-active;
+
+            &
+            :after {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
            &:after {
```

File: src/Multiselect.vue [hunk 21]
```diff
@@ -143,689 +143,738 @@
...
+    .multiselect__option--disabled.multiselect__option--highlight {
+        background: $multiselect-color-background-group-disabled;
+    }
+
+    .multiselect__option--group-selected.multiselect__option--highlight {
+        background: $multiselect-color-group;
+        color: $multiselect-color-text-active;
+
+        &
+        :after {
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
        &:after {
```

File: src/Multiselect.vue [hunk 22]
```diff
@@ -143,689 +143,738 @@
...
+    }
+
+    .multiselect__content-wrapper {
+        position: absolute;
+        display: block;
+        background: $multiselect-color-background;
+        width: 100%;
+        max-height: 240px;
+        overflow: auto;
+        border: 1px solid #e8e8e8;
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
        border: 1px solid $multiselect-color-item-border
```

File: src/Multiselect.vue [hunk 23]
```diff
@@ -143,689 +143,738 @@
...
+    }
+
+    .multiselect--above .multiselect__content-wrapper {
+        bottom: 100%;
+        border-bottom-left-radius: 0;
+        border-bottom-right-radius: 0;
+        border-top-left-radius: 5px;
+        border-top-right-radius: 5px;
+        border-bottom: none;
+        border-top: 1px solid #e8e8e8;
```
Comment (mathieutu) [inline src/Multiselect.vue]:
```suggestion
         border-top: 1px solid $multiselect-color-item-border;
```

**Comment by mathieutu** [comment]
Hum, I didn't know that github would make a commit for every suggestion, sorry for that.. 😕

**Comment by eFrane** [comment]
Yeah no, that was my fault. But I will rebase this thing anyway before we merge it.

**Comment by assembledadam** [comment]
Yeah, nice! 

Spammmm

**Comment by eFrane** [comment]
Sorry again! It's not always easy to Github.

**Comment by kosmiq** [comment]
A thought is that perhaps we should be able to define the classname used for multi-select. This allows users to change the CSS classname from ".multiselect*" to whatever they prefer to avoid clashes with existing/other components.

This would be accomplished by setting up a variable like this:
`$vuemultiselect-selector: 'multiselect' !default;`

And then we would use in the SCSS like this:
```
.#{$vuemultiselect-selector}__spinner {
...
}
```

And this would create classes like before if not defined, and with a new class name base if defined.

We would however need the ability to define on the component what class to use. I'm not familiar enough with VueJS to accomplish that yet. But I could setup the SCSS tooling for it.

**Comment by eFrane** [comment]
This appears to be dead.

**Pull request closed without merging**